src/components/MessageManager/MessageManager.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from "react";
+import "./MessageManager.css";
+import MessagingWindow from "../MessagingWindow/MessagingWindow";
+
+import Firebase from "firebase";
+
+class MessageManager extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      friends: [],
+      friendClicked: {},
+      user: this.props.user
+    };
+  }
+
+  componentDidMount() {
+    this.friendsRef = Firebase.database().ref(
+      "users/" + this.props.user + "/friends/"
+    ); //reference to friends
+    this.friendsRef.on("value", snapshot => {
+      this.setState(
+        {
+          friends: snapshot.val() //gives complete list of friends
+        },
+        () => {
+          let newClickedList = {};
+          Object.keys(this.state.friends).map(
+            (f, i) => (newClickedList[f] = false)
+          );
+          this.setState({
+            friendClicked: newClickedList
+          });
+        }
+      );
+    });
+  }
+
+  handleFriendClick = f => {
+    let prevFriendClicked = this.state.friendClicked;
+    prevFriendClicked[f] = !this.state.friendClicked[f];
+    this.setState({ friendClicked: prevFriendClicked });
+  };
+
+  render() {
+    return (
+      <div>
+        <h1>Temporary Message Manager</h1>
+
+        {/* TODO: Display list of friends */}
+        {console.log(this.state)}
+        {Object.keys(this.state.friends).map((f, i) => (
+          <div key={i}>
+            <button
+              className="friend-name"
+              onClick={() => this.handleFriendClick(f)}
```

Comment:
if i remember correctly, theres no point of putting the function in an anonymous function inline like this if its already defined as an anonymous function above

Reply:
The reason I did it this way is because the function takes in an argument! So if I just did `onClick={this.handleFriendClick(f)}`, it would repeatedly call the function and exceed max depth! If the function did not take any arguments, your point would be completely valid, for I could just say `onClick={this.handleFriendClick}`. But notice the difference in behavior if I wrote `onClick={this.handleFriendClick()}`.

So in this case, I put a dummy anonymous function outside, to ensure that the function is called only on click as follows: `onClick={() => this.handleFriendClick(f)}`